monoio-http/src/h1/codec/encoder.rs

```
@@ -275,22 +291,24 @@ where
                     } else {
                         // the data length is small, we copy it to avoid too many
                         // syscall(head and body will be sent together).
-                        FixedBodyEncoder.encode(&data, &mut self.buf)?;
+                        let slice = unsafe {
+                            std::slice::from_raw_parts(data.read_ptr(), data.bytes_init())
+                        };
+                        FixedBodyEncoder.encode(slice, &mut self.buf)?;
                     }
                 }
-                Payload::Stream(mut p) => {
+                StreamHint::Stream => {
                     // set special header
                     HeadEncoder::set_length_header(header_map, Length::Chunked);
                     // encode head to buffer
                     HeadEncoder
                         .encode(head, &mut self.buf)
                         .map_err(Into::into)?;
 
-                    while let Some(data_result) = p.next().await {
-                        let data = data_result?;
-                        write!(self.buf, "{:X}\r\n", data.len())
+                    while let Ok(Some(data)) = payload.data().await {
```

The `while let Ok(Some(data))` hides error.